src/Dialect/ONNX/ONNXOps/NN/Normalization.cpp

```diff
@@ -149,3 +149,124 @@ LogicalResult ONNXInstanceNormalizationOp::verify() {
 }
 
 // TODO: should there be a shape inference for this one?
+
+//===----------------------------------------------------------------------===//
+// LayerNormalization
+//===----------------------------------------------------------------------===//
+
+LogicalResult ONNXLayerNormalizationOp::verify() {
+  auto operandAdaptor = ONNXLayerNormalizationOpAdaptor(*this);
+
+  // Get attributes.
+  int64_t axis = getAxis();
+
+  // Get operands.
+  Value X = operandAdaptor.getX();
+  Value scale = operandAdaptor.getScale();
+  Value B = operandAdaptor.getB();
+
+  // Check X.
+  if (!hasShapeAndRank(X)) {
+    // Won't be able to do any checking at this stage.
+    return success();
+  }
+  ShapedType XType = X.getType().cast<ShapedType>();
+  ArrayRef<int64_t> XShape = XType.getShape();
+  int64_t XRank = XShape.size();
+  Type XElementType = XType.getElementType();
+
+  // Axis attribute (if specified) must be in the range [-r,r), where r =
+  // rank(input).
+  if (axis < -XRank || axis >= XRank)
+    return emitOpError("axis must be in [-r, r) range]");
+  if (axis < 0)
+    axis += XRank;
+
+  // Check bias B.
+  if (hasShapeAndRank(B)) {
+    // Can check at this stage.
+    ShapedType bType = B.getType().cast<ShapedType>();
+    ArrayRef<int64_t> bShape = bType.getShape();
+    SmallVector<int64_t> bcastShape;
+    if (!OpTrait::util::getBroadcastedShape(XShape, bShape, bcastShape))
+      emitOpError(
+          "LayerNormalization op with incompatible B shapes (broadcast)");
+    if ((int64_t)bcastShape.size() != XRank)
+      emitOpError("LayerNormalization op with incompatible B shapes "
+                  "(unidirectional broadcast)");
+    if (bType.getElementType() != XElementType)
+      emitOpError("LayerNormalization op with incompatible B type");
+  }
+
+  // Check scale.
+  if (hasShapeAndRank(scale)) {
+    // Can check at this stage.
+    ShapedType scaleType = scale.getType().cast<ShapedType>();
+    ArrayRef<int64_t> scaleShape = scaleType.getShape();
+    SmallVector<int64_t> bcastShape;
```

Comment:
it would be consistent if we name it `scaleCastShape` instead.

Reply:
`beast` stands for broadcast. I know changed it to `BBroadcastShape` and `scaleBroadcstChape` to be more explicit. Thanks for the catch.

Reply:
I see, I though b was B operand.